app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,72 @@
+class ShipmentsController < ApplicationController
+
+  def shipment
+    submission_errors = submission_error_handling(params)
+
+    if submission_errors
+      return submission_errors
+    else
+      origin = ActiveShipping::Location.new(country: params[:origin][:country], state: params[:origin][:state], city: params[:origin][:city], zip: params[:origin][:zip])
```

Comment:
This line is long and difficult to read on github. It's good to know a tool's limits, and for work with tools that don't [soft wrap](http://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap) text I recommend breaking each parameter onto its own line for readability. Here's what it might look like:

<pre>origin = ActiveShipping::Location.new(
      country: params[:origin][:country],
        state: params[:origin][:state],
         city: params[:origin][:city],
          zip: params[:origin][:zip])</pre>